QEfficient/loader/loader_factory.py

```diff
@@ -0,0 +1,121 @@
+# -----------------------------------------------------------------------------
+#
+# Copyright (c) 2024 Qualcomm Innovation Center, Inc. All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+#
+# ----------------------------------------------------------------------------
+
```

Comment:
can you add details about the design pattern used here, with some depiction of class inheritence

Reply:
Added.